apps/meteor/app/api/server/v1/assets.ts

```diff
@@ -1,13 +1,18 @@
+import { Settings } from '@rocket.chat/models';
 import { isAssetsUnsetAssetProps } from '@rocket.chat/rest-typings';
 
-import { RocketChatAssets, setAsset, unsetAsset, refreshClients } from '../../../assets/server';
+import { RocketChatAssets, refreshClients } from '../../../assets/server';
+import { notifyOnSettingChangedById } from '../../../lib/server/lib/notifyListener';
 import { settings } from '../../../settings/server';
 import { API } from '../api';
 import { getUploadFormData } from '../lib/getUploadFormData';
 
 API.v1.addRoute(
 	'assets.setAsset',
-	{ authRequired: true },
+	{
+		authRequired: true,
+		permissionsRequired: ['manage-assets'],
```

Comment:
Would it be possible to add the (unused) validation `isAssetsSetAssetProps`?

Reply:
I'm afraid not, in this case the body type is not a json, and needs to be built first, with our current technology our parser is not smart enough to do this for us :/